purgatory/string.c: Add const to the memcmp function

Index: kexec-tools/purgatory/string.c
===================================================================
--- kexec-tools.orig/purgatory/string.c
+++ kexec-tools/purgatory/string.c
@@ -36,9 +36,9 @@ void* memcpy(void *dest, const void *src
 }
 
 
-int memcmp(void *src1, void *src2, size_t len)
+int memcmp(const void *src1, const void *src2, size_t len)
 {
-	unsigned char *s1, *s2;
+	const unsigned char *s1, *s2;
 	size_t i;
 	s1 = src1;
 	s2 = src2;
Index: kexec-tools/purgatory/include/string.h
===================================================================
--- kexec-tools.orig/purgatory/include/string.h
+++ kexec-tools/purgatory/include/string.h
@@ -6,7 +6,7 @@
 size_t strnlen(const char *s, size_t max);
 void* memset(void* s, int c, size_t n);
 void* memcpy(void *dest, const void *src, size_t len);
-int memcmp(void *src1, void *src2, size_t len);
+int memcmp(const void *src1, const void *src2, size_t len);
 
 
 #endif /* STRING_H */
