From ccc416fba4a148c41d108ae74c68afa8b9908540 Mon Sep 17 00:00:00 2001
From: Daniel Dragomir <Daniel.Dragomir@windriver.com>
Date: Mon, 18 Oct 2021 13:00:23 +0300
Subject: [PATCH] bootlogd: add suport for custom ttyAMA console

Signed-off-by: Daniel Dragomir <Daniel.Dragomir@windriver.com>
---
 src/bootlogd.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/bootlogd.c b/src/bootlogd.c
index 8b07903..a2a40bd 100644
--- a/src/bootlogd.c
+++ b/src/bootlogd.c
@@ -85,6 +85,7 @@ struct consdev {
 	char	*dev2;
 } consdev[] = {
 	{ "ttyB",	"/dev/ttyB%s",		NULL		},
+	{ "ttyAMA",	"/dev/ttyAMA%s",	"/dev/ttama/%s"	},
 	{ "ttySC",	"/dev/ttySC%s",		"/dev/ttsc/%s"	},
 	{ "ttyS",	"/dev/ttyS%s",		"/dev/tts/%s"	},
 	{ "tty",	"/dev/tty%s",		"/dev/vc/%s"	},
@@ -96,7 +97,7 @@ struct consdev {
  *	Devices to try as console if not found on kernel command line.
  *	Tried from left to right (as opposed to kernel cmdline).
  */
-char *defcons[] = { "tty0", "hvc0", "ttyS0", "ttySC0", "ttyB0", NULL };
+char *defcons[] = { "ttyAMA0", "tty0", "hvc0", "ttyS0", "ttySC0", "ttyB0", NULL };
 
 /*
  *	Catch signals.
-- 
2.17.1

